library.rb

```diff
@@ -49,4 +52,102 @@ def list_rentals
       end
     end
   end
+
+  def parse_people_json(data_array)
+    people_data = []
+    rental_data = []
+    data_array.each do |data|
+      data.rentals.each do |rental|
+        rental_data.push({ Date: rental.date, BookId: rental.book.id, PersonId: rental.person.id })
+      end
+      case data.class.to_s
+      when 'Student'
+        people_data.push({ Class: data.class, Name: data.name, ID: data.id, Age: data.age, Rentals: rental_data })
+      when 'Teacher'
+        people_data.push({ Class: data.class, Name: data.name, ID: data.id, Age: data.age,
+                           Specialization: data.specialization, Rentals: rental_data })
+      end
+      rental_data = []
+    end
+    JSON.generate(people_data)
+  end
+
+  def parse_book_json(data_array)
+    book_data = []
+    rental_data = []
+    data_array.each do |data|
+      data.rentals.each do |rental|
+        rental_data.push({ Date: rental.date, BookId: rental.book.id, PersonId: rental.person.id })
+      end
+      book_data.push({ Title: data.title, Author: data.author, ID: data.id, Rentals: rental_data })
+      rental_data = []
+    end
+    JSON.generate(book_data)
+  end
+
+  def parse_rental_json(data_array)
+    rental_data = []
+    data_array.each do |data|
+      rental_data.push({ Date: data.date, BookId: data.book.id, PersonId: data.person.id })
+    end
+    JSON.generate(rental_data)
+  end
+
+  def read_people()
+    return [] unless File.exist?('person.json')
+
+    file = File.open('person.json')
+    file_data = file.read if file
+    people_data = JSON.parse(file_data)
+    people = []
+    people_data.each do |data, _index|
+      case data['Class']
+      when 'Student'
+        people.push(Student.new(data['ID'], data['Age'], data['Name']))
+      when 'Teacher'
+        people.push(Teacher.new(data['ID'], data['Age'], data['Specialization'], data['Name']))
+      end
+    end
+    @people = people
+  end
+
+  def read_books()
+    return [] unless File.exist?('books.json')
+
+    file = File.open('books.json')
+    file_data = file.read if file
+    books_data = JSON.parse(file_data)
+    books = []
+    books_data.each do |data, _index|
+      books.push(Book.new(data['ID'], data['Title'], data['Author']))
+    end
+    @book_list = books
+  end
+
+  def read_rentals()
+    return [] unless File.exist?('rentals.json')
+
+    file = File.open('rentals.json')
+    file_data = file.read if file
+    rentals_data = JSON.parse(file_data)
+    rentals = []
+    rentals_data.each do |data|
+      rental_book = @book_list.find { |book| book.id == data['BookId'] }
+      rental_person = @people.find { |person| person.id == data['PersonId'] }
+      rentals.push(Rental.new(data['Date'], rental_book, rental_person))
+    end
+    @rentals = rentals
+  end
```

Comment:
- [ ] On startup of the app, load all data from files. Currently, previous data is not preserved. Check the Image below. A student is added but the list is returned empty.
![Uploading Screenshot from 2021-11-25 09-14-40.png…]()

Reply:
Thank you mke2111, now this bug 🐛 it's fixed 😄